src/Diagrams/TwoD/Layout/Constrained.hs

```diff
@@ -620,12 +622,20 @@ layout
   :: (Monoid' m, Hashable n, Floating n, RealFrac n, Show n)
   => (forall s. Constrained s b n m a)
   -> QDiagram b V2 n m
-layout constr =
+layout constr = snd $ runLayout constr
+
+-- | Like 'layout', but also allows the caller to retrieve the monadic value
```

Comment:
Minor nitpick, I don't like the terminology of calling `a` a "monadic value", or of saying that things are "inside the Constrained monad".  Could we instead say something like "...retrieve the result of the 'Constrained' computation"?